vm/src/obj/objbytes.rs

```diff
@@ -0,0 +1,55 @@
+use super::super::pyobject::{
+    AttributeProtocol, FromPyObjectRef, PyContext, PyFuncArgs, PyObjectKind, PyObjectRef, PyResult,
+    TypeProtocol,
+};
+use super::super::vm::VirtualMachine;
+use super::objint;
+use super::objlist;
+use super::objtype;
+// Binary data support
+
+// Fill bytes class methods:
+pub fn init(context: &PyContext) {
+    let ref bytes_type = context.bytes_type;
+    bytes_type.set_attr("__init__", context.new_rustfunc(bytes_init));
+    bytes_type.set_attr("__str__", context.new_rustfunc(bytes_str));
+}
+
+// __init__ (store value into objectkind)
+fn bytes_init(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(
+        vm,
+        args,
+        required = [(zelf, Some(vm.ctx.bytes_type())), (arg, None)]
+    );
+    let val = if objtype::isinstance(arg.clone(), vm.ctx.list_type()) {
+        objlist::get_elements(arg.clone())
+            .into_iter()
+            .map(|e| objint::get_value(&e) as u8)
```

Comment:
`objint::get_value` should only be used when we are _sure_ that we have an int (because it will panic if we don't); as this is data fed to us by a function call, we should instead be checking that we have ints and raising a TypeError if we don't.

For reference, in CPython:
```
>>> bytes([object()])
TypeError: 'object' object cannot be interpreted as an integer
```